test/conftest.py

```diff
@@ -21,9 +34,53 @@ def app() -> Flask:
                 'client_id': 'server_client_id',
                 'client_secret': 'server_client_secret',
             }
-        }
+        },
     })
 
+    ctx = app.app_context()
+    ctx.push()
```

Comment:
some comments on what global-state-restoring I suppose is performed here?

Reply:
Based on http://alexmic.net/flask-sqlalchemy-pytest/. Need to go through this part a bit better and work out what's happening.